### PR TITLE
Makes hypnosis victims more visible to admins

### PR DESCRIPTION
**Pull request by dragomagol**
## About The Pull Request
image
Gives hypnotized victims a visible antag hud for admins, puts them in the Check Antags panel, and makes their objectives visible in the Traitor Panel and on the round end report. 

## Why It's Good For The Game
Right now, victims of hypnosis are very difficult to visually identify as an admin. This not only distinguishes them easily but makes their objective easily visible too. 

## Changelog

:cl:
admin: hypnotized players now have a visible antag hud, are visible in the check antag panel, and have objectives visible on the traitor panel
expansion: hypnotized victims and their objectives are now visible on the round end report
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
does removing the trauma remove the antag status?

**Comment by Iamgoofball** [comment]
Talk about this one with admins, this means hypnotized players are antags and now fall under the "antags can do what they want" rules.

**Comment by dragomagol** [comment]
This falls under the same purview as brainwashed victims and abductees. Neither of those override Rule 4. This is purely a mechanical change because this is how we classify those others.

**Comment by SoSMaster9000** [comment]
Oh my god, THANK YOU!

**Comment by dragomagol** [comment]
> 
> 
> does removing the trauma remove the antag status?

It did not, now it does

File: code/modules/antagonists/hypnotized/hypnotized.dm [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+	antagpanel_category = "Other"
+	show_name_in_check_antagonists = TRUE
+
+	silent = TRUE //not actually silent, because greet will be called by the trauma anyway.
+	var/datum/brain_trauma/hypnosis/trauma
+
+/datum/antagonist/hypnotized/Destroy()
+	if(trauma)
+		qdel(trauma)
+	. = ..()
```
Comment (Mothblocks) [inline code/modules/antagonists/hypnotized/hypnotized.dm:17]:
```suggestion
	QDEL_NULL(trauma)
	return ..()
```

File: code/modules/antagonists/hypnotized/hypnotized.dm [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+	job_rank = ROLE_HYPNOTIZED
+	roundend_category = "hypnotized victims"
+	antag_hud_name = "brainwashed"
+	ui_name = "AntagInfoBrainwashed"
+	show_in_antagpanel = TRUE
+	antagpanel_category = "Other"
+	show_name_in_check_antagonists = TRUE
+
+	silent = TRUE //not actually silent, because greet will be called by the trauma anyway.
+	var/datum/brain_trauma/hypnosis/trauma
```
Comment (Mothblocks) [inline code/modules/antagonists/hypnotized/hypnotized.dm:12]: Auto-doc this, since you use it elsewhere.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/modules/antagonists/hypnotized/hypnotized.dm (hunk 2 above)
Comment (Mothblocks) [reply code/modules/antagonists/hypnotized/hypnotized.dm:12]: You didn't auto-doc this, I think you just auto-docced the other thing @dragomagol

Comment (dragomagol) [reply code/modules/antagonists/hypnotized/hypnotized.dm:14]: Actually resolved this time B)

**Pull request merged**